docs/HowToHashiCorpVault.md

```diff
@@ -0,0 +1,278 @@
+In a production deployment, you would typically configure additional security settings, but it serves as a good example while you test and familiarize yourself with this feature.
+
+```
+kubectl exec -it -n vault vault-0 -- sh
+$ cat <<EOF > /home/vault/nuodb-policy.hcl
+path "nuodb.com*" {
+  capabilities = ["read"]
+}
+EOF
+
+$ vault policy write nuodb-policy /home/vault/nuodb-policy.hcl
+```
+
+### Enable Kubernetes Integration
+
+The next step is to enable the [Vault Kubernetes Auth](https://www.vaultproject.io/docs/auth/kubernetes) method.
+The `nuodb-policy` created above in step [Create Vault Policy](###create-vault-policy) will get attached to the `nuodb` namespace.
+
+```
+kubectl exec -it -n vault vault-0 -- sh
+$ vault auth enable kubernetes
+
+$ vault write auth/kubernetes/config \
+   token_reviewer_jwt="$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
+   kubernetes_host=https://${KUBERNETES_PORT_443_TCP_ADDR}:443 \
+   kubernetes_ca_cert=@/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+
+$ vault write auth/kubernetes/role/nuodb \
+   bound_service_account_names=nuodb \
+   bound_service_account_namespaces=nuodb \
+   policies=nuodb-policy \
+   ttl=1h
+```
+
+### Create a NuoDB Secrets Engine
+
+The final step is to create a [Secrets KV Engine](https://www.vaultproject.io/docs/secrets/kv/index.html) to store the NuoDB keys.
+All keys will be base64 encoded.
+
+```
+kubectl exec -it -n vault vault-0 -- sh
+$ vault secrets enable -version=2 -path=nuodb.com kv
+```
+
+## Adding NuoDB Keys to HashiCorp Vault
+
+In the previous step [Generating NuoDB Keys](##generating-nuodb-keys) we generated a set of keys required to run NuoDB.
+We saved those keys in `/tmp/keys`.
+In this step, we will save these keys in HashiCorp Vault.
+
+```
+kubectl exec -it -n vault vault-0 -- \
+    vault kv put nuodb.com/TLS \
+    tlsClientPEM=`cat /tmp/keys/nuocmd.pem | base64` \
+    tlsCACert=`cat /tmp/keys/ca.cert | base64` \
+    tlsKeyStorePassword=changeIt \
+    tlsTrustStorePassword=changeIt \
+    tlsKeyStore=`cat /tmp/keys/nuoadmin.p12 | base64` \
+    tlsTrustStore=`cat /tmp/keys/nuoadmin-truststore.p12 | base64`
+```
+
+## Configuration of NuoDB
+
+HashiCorp Vault uses annotations to identify pods that require a Vault Agent.
+For more information please consult the [Template Documentation](https://www.vaultproject.io/docs/platform/k8s/injector/annotations#vault-hashicorp-com-agent-inject-template)
+
+### Annotations for the NuoDB Admin Tier
+
+The NuoDB admin tier requires all four key files. 
+For convenience, we list them here again:
+- `nuodb-keystore.p12` containing the X509 that identifies the admin;
+- `nuodb-truststore.p12` usually containing the root CA and the primordial admin user;
+- `ca.cert` containing the public certificate of the Certificate Authority;
+- `nuocmd.pem` which is the private/public keypair for the primordial admin user.
+
+Both the keystore and the truststore PKCS12 files are password protected.
```

Comment:
Technically, the password is only used to verify the truststore (to make sure certificates aren't smuggled in).